src/main/middleware/Authentication.ts

```diff
@@ -0,0 +1,35 @@
+import { NextFunction } from 'express';
+import { Request, Response } from 'express';
+import { auth } from 'firebase-admin/lib/auth';
+import Auth = auth.Auth;
+
+/**
+ * Middleware function to check if the request has correct client header using firebase-admin auth
+ */
+export class FirebaseAuth {
+  private admin: Auth;
+
+  constructor(firebaseAdmin: Auth) {
+    this.admin = firebaseAdmin;
+  }
+
+  isAuthorizedForProtection = async (
```

Comment:
Nice! Really like how simple it looks now. Could we add some documentation for this function?

Reply:
I added!